packages/postgres-database/tests/test_models_products_prices.py

```diff
@@ -0,0 +1,153 @@
+    connection: SAConnection, fake_product: RowProxy
+):
+    # negative price not allowed
+    with pytest.raises(CheckViolation) as exc_info:
+        await connection.execute(
+            products_prices.insert().values(
+                product_name=fake_product.name,
+                usd_per_credit=-100,  # <----- NEGATIVE
+                authorized_by="PO Mr X",
+            )
+        )
+
+    assert exc_info.value
+
+    # zero price is allowed
+    await connection.execute(
+        products_prices.insert().values(
+            product_name=fake_product.name,
+            usd_per_credit=0,  # <----- ZERO
+            authorized_by="PO Mr X",
+        )
+    )
+
+
+async def test_delete_price_constraints(
+    connection: SAConnection, fake_product: RowProxy
+):
+    # products_prices
+    await connection.execute(
+        products_prices.insert().values(
+            product_name=fake_product.name,
+            usd_per_credit=10,
+            authorized_by="PO Mr X",
+        )
+    )
+
+    # should not be able to delete a product w/o deleting price first
+    with pytest.raises(ForeignKeyViolation) as exc_info:
+        await connection.execute(products.delete())
+
+    assert "delete" in f"{exc_info.value}"
+
+    # this is the correct way to delete
+    await connection.execute(products_prices.delete())
+    await connection.execute(products.delete())
+
+
+async def test_get_product_latest_price_or_none(
+    connection: SAConnection, fake_product: RowProxy
+):
+    # undefined product
+    assert (
+        await get_product_latest_price_or_none(connection, product_name="undefined")
+        is None
+    )
+
+    # defined product but undefined price
+    assert (
+        await get_product_latest_price_or_none(
+            connection, product_name=fake_product.name
+        )
+        is None
+    )
+
+
+async def test_price_history_of_a_product(
+    connection: SAConnection, fake_product: RowProxy
+):
+    # initial price
+    await connection.execute(
+        products_prices.insert().values(
+            product_name=fake_product.name,
+            usd_per_credit=1,
+            authorized_by="PO Mr X",
+        )
+    )
+
+    # new price
+    await connection.execute(
+        products_prices.insert().values(
+            product_name=fake_product.name,
+            usd_per_credit=2,
+            authorized_by="Update by Mr X",
+        )
+    )
+
+    result = await connection.execute(
+        products_prices.select()
+        .where(products_prices.c.product_name == fake_product.name)
+        .order_by(sa.desc(products_prices.c.created))
+    )
+    rows = await result.fetchall()
+    assert rows
+    assert len(rows) == 2
+    assert rows[0].usd_per_credit == 2, "Newest first"
+    assert rows[1].usd_per_credit == 1
+
+    # latest is 2 USD!
+    assert (
+        await get_product_latest_price_or_none(
+            connection, product_name=fake_product.name
+        )
+        == 2
+    )
```

Comment:
I think is is sufficient. Since you know that you just added the latest price which is equal to 2 and is different from the one before it.